sway.5.txt

```diff
@@ -175,3 +179,34 @@ Commands
 **workspace** <name> output <output>::
 	Specifies that the workspace named _name_ should appear on the specified
 	_output_.
+
+Criteria
+--------
+
+A criteria is a string in the form of e.g.:
+
+	[class="[Rr]egex.*" title="some title"]
+
+The string contains one or more (space separated) attribute/value pairs and they
+are used by some commands filter which views to execute actions on. All attributes
+must match for the criteria string to match.
+
+Currently supported attributes:
+
+**class**::
+	Compare value against the window class. Can be a regular expression. If value
+	is _focused_ then the window class must be the same as that of the currently
+	focused window.
+
+**id**::
+	Compare against the app id. Must be a normal string.
```

Comment:
From the userguide:

> Compares the X11 window ID, which you can get via xwininfo for example.

The window id is an arbitrarily chosen number, much like a pointer in C. It’s not meaningful to do a regular expression match on the id, hence i3 just doesn’t support it at all.

For what it’s worth, i3 parses the id into an xcb_window_t, which is just a uint32_t underneath. Hence, it’s not actually a string, but rather a number (in typically either decimal or hexadecimal). This also makes matching more efficient: in i3, we compare two uint32_ts, whereas, if you were to implement it as a regexp, you’d need to first sprintf() the id into a string, then apply the regexp.